CONTRIBUTING.md

```diff
@@ -1,24 +1,38 @@
 # How to Contribute
 
-Contributions to libCEED are welcome.
+Contributions to libCEED are encouraged.
 
 Please use a pull request to the appropriate branch ('maint' for
 backward-compatible bug fixes for the last stable release, 'master' for
 new features and everything else).
 
-Please make your commits well-organized and atomic, using `git rebase
---interactive` as needed.  Open an issue or RFC pull request to discuss
-any significant changes before investing time.  Check that the tests
-pass using `make prove` or `make test`.  If adding a new feature, please
-add or extend a test so that it is tested.
+Please make your commits well-organized and
```

Comment:
I think it will be useful to split recommendations from requirements.

For example, I will consider the atomic commits a recommendation, while the pass of  `make prove-all` a requirement.

Reply:
Yeah, we could split it into two subsections. Regarding atomicity, I think it's a matter of degrees. A single commit for a new backend is totally fine, but if we saw one commit that made sweeping formatting or naming changes in the same commit as a significant feature or bug fix, I'd expect to ask for it to be broken apart. I think of the guideline as "please make a reasonable attempt" on these subjective matters -- perhaps just rephrasing would be sufficient.